src/modern/class/__tests__/ReactES6Class-test.js

```diff
@@ -65,6 +65,129 @@ describe('ReactES6Class', function() {
     test(<Foo bar="bar" />, 'DIV', 'bar');
   });
 
+  it('renders based on state using initial values in this.props', function() {
+    class Foo extends React.Component {
+      constructor(props) {
+        super(props);
+        this.state = { bar: this.props.initialValue };
+      }
+      render() {
+        return <span className={this.state.bar} />;
+      }
+    }
+    test(<Foo initialValue="foo" />, 'SPAN', 'foo');
+  });
+
+  it('renders based on state using props in the constructor', function() {
+    class Foo extends React.Component {
+      constructor(props) {
+        this.state = { bar: props.initialValue };
+      }
+      changeState() {
+        this.setState({ bar: 'bar' });
+      }
+      render() {
+        if (this.state.bar === 'foo') {
+          return <div className="foo" />;
+        }
+        return <span className={this.state.bar} />;
+      }
+    }
+    var instance = test(<Foo initialValue="foo" />, 'DIV', 'foo');
+    instance.changeState();
+    test(<Foo />, 'SPAN', 'bar');
+  });
+
+  it('renders only once when setting state in componentWillMount', function() {
+    var renderCount = 0;
+    class Foo extends React.Component {
+      constructor(props) {
+        this.state = { bar: props.initialValue };
+      }
+      componentWillMount() {
+        this.setState({ bar: 'bar' });
+      }
+      render() {
+        renderCount++;
+        return <span className={this.state.bar} />;
+      }
+    }
+    test(<Foo initialValue="foo" />, 'SPAN', 'bar');
+    expect(renderCount).toBe(1);
+  });
+
+  it('should throw with non-object in the initial state property', function() {
+    [['an array'], 'a string', 1234].forEach(function(state) {
+      class Foo {
+        constructor() {
+          this.state = state;
+        }
+        render() {
+          return <span />;
+        }
+      }
+      expect(() => test(<Foo />, 'span', '')).toThrow(
+        'Invariant Violation: Foo.state: ' + 
+        'must be set to an object or null'
+      );
+    });
+  });
+
+  it('should render with null in the initial state property', function() {
+    class Foo extends React.Component {
+      constructor() {
+        this.state = null;
```

Comment:
Is `this.state` undefined before this or is it `null` already? Worth adding a test for?